lib/msf/core/web_services/authentication/strategies/user_password.rb

```diff
@@ -16,15 +16,19 @@ class UserPassword < Warden::Strategies::Base
       # Check if request contains valid data and should be authenticated.
       # @return [Boolean] true if strategy should be run for the request; otherwise, false.
       def valid?
-        params['username'] && params['password']
+        body = JSON.parse(request.body.read, symbolize_names: true)
+        request.body.rewind # Reset the StringIO buffer so any further consumers can read the body
```

Comment:
Would it be a good idea to have a begin-ensure block around the operations of the method and move the `request.body.rewind` to the `ensure` as a fail-safe should the JSON received cause an error on `JSON.parse`?
```
begin
  body = JSON.parse(request.body.read, symbolize_names: true)
  body[:username] && body[:password]
ensure
  request.body.rewind # Reset the StringIO buffer so any further consumers can read the body
end
```